src/core/symbology/qgsfillsymbollayer.cpp

```diff
@@ -4047,12 +4056,26 @@ void QgsRandomMarkerFillSymbolLayer::render( QgsRenderContext &context, const QV
     context.painter()->setClipPath( path );
   }
 
+
   int count = mPointCount;
   if ( mDataDefinedProperties.isActive( QgsSymbolLayer::PropertyPointCount ) )
   {
     context.expressionContext().setOriginalValueVariable( count );
     count = mDataDefinedProperties.valueAsInt( QgsSymbolLayer::PropertyPointCount, context.expressionContext(), count );
   }
+
+  if ( mCountMethod == DensityBasedCount )
+  {
+    double densityArea = mDensityArea;
+    if ( mDataDefinedProperties.isActive( QgsSymbolLayer::PropertyDensityArea ) )
+    {
+      context.expressionContext().setOriginalValueVariable( densityArea );
+      densityArea = mDataDefinedProperties.valueAsDouble( QgsSymbolLayer::PropertyPointCount, context.expressionContext(), densityArea );
+    }
+    densityArea = context.convertToPainterUnits( densityArea, mDensityAreaUnit, mDensityAreaUnitScale );
```

Comment:
This method converts distances, not areas. I think you'll need to adapt this accordingly.

Reply:
Hmm, so taking the root square value, then converting, then taking the square value of that?